test/run.js

```diff
@@ -1,26 +1,39 @@
 'use strict';
 
-var bulk = require('bulk-require');
-var test = require('tape');
+var path = require('path');
+var jest = require('jest');
 var algolia = require('algoliasearch');
+var staticJestConfig = require('../jest.config');
 
-bulk(__dirname, ['spec/**/*.js']);
+var enableIntegrationTest =
+  process.env.ONLY_UNIT !== 'true' &&
+  process.env.INTEGRATION_TEST_API_KEY &&
+  process.env.INTEGRATION_TEST_APPID;
 
-if (process.env.ONLY_UNIT !== 'true' && process.env.INTEGRATION_TEST_API_KEY && process.env.INTEGRATION_TEST_APPID) {
-  // usage: INTEGRATION_TEST_APPID=$APPID INTEGRATION_TEST_API_KEY=$APIKEY npm run dev
-  bulk(__dirname, ['integration-spec/**/*.js']);
+var projectsRootPaths = [path.resolve(__dirname, '..')];
+var dynamicJestConfig = Object.assign({}, staticJestConfig, {
+  maxWorkers: 4,
+  setupFilesAfterEnv: staticJestConfig.setupFilesAfterEnv || []
+});
 
-  test.onFinish(cleanupIntegration);
+if (enableIntegrationTest) {
+  dynamicJestConfig.testMatch.push('<rootDir>/test/integration-spec/**/*.[jt]s?(x)');
+  dynamicJestConfig.setupFilesAfterEnv.push(path.resolve(__dirname, '..', 'jest.setup.js'));
 }
 
-function cleanupIntegration() {
-  console.log('Deleting all indices');
-  var client = algolia(process.env.INTEGRATION_TEST_APPID, process.env.INTEGRATION_TEST_API_KEY);
-  var maybeIndices = client.listIndexes();
-  maybeIndices.then(content => {
-    content.items
-      .map(i => i.name)
-      .filter(n => n.indexOf('_travis-algoliasearch-helper') !== -1)
-      .forEach(n => client.deleteIndex(n));
-  });
-}
+jest.runCLI(dynamicJestConfig, projectsRootPaths).then(function(response) {
+  if (!response.results.success) {
+    process.exitCode = response.globalConfig.testFailureExitCode;
+  }
+
+  if (enableIntegrationTest) {
+    var client = algolia(process.env.INTEGRATION_TEST_APPID, process.env.INTEGRATION_TEST_API_KEY);
+
+    client.listIndexes().then(content => {
+      content.items
+        .map(i => i.name)
```

Comment:
these likely might need to be in regular functions? or I guess it is fine since it's only ran in node

Reply:
Yes indeed, it was already like that. All the tests could be wrote with modern JavaScript, but we have to update the linter to support some modern syntax like spread operator.